helpers.py

```diff
@@ -21,9 +21,57 @@
 import subprocess
 import time
 
-from fabric import Config
-from fabric import Connection
-from paramiko import SSHException
+
+class Result:
+    def __init__(self, stdout, stderr, exited):
+        self.stdout = stdout
+        self.stderr = stderr
+        self.exited = exited
+        self.return_code = exited
+
+
+class Connection:
+    def __init__(self, host, user, port, connect_timeout):
+        self.host = host
+        self.user = user
+        self.port = port
+        self.connect_timeout = connect_timeout
+
+    def __enter__(self):
+        return self
+
+    def run(self, command, warn=False, hide=True, echo=False, popen=False):
+        if echo:
+            print(command)
+
+        if popen:
+            return subprocess.Popen(command)
+        else:
+            try:
+                proc = subprocess.run(
+                    command, check=not warn, capture_output=True, shell=True
+                )
+                returncode = proc.returncode
+
+            except subprocess.CalledProcessError as e:
+                returncode = e.returncode
+                if returncode != 255:
+                    raise
+
+            if returncode == 255:
+                raise ConnectionError(f"Could not connect using command '{command}'")
+
+            stdout = proc.stdout.decode()
+            stderr = proc.stderr.decode()
+
+            if not hide:
+                print(stdout)
+                print(stderr)
+
+            return Result(stdout, stderr, returncode)
```

Comment:
The code is in essence the same as in `integration` repository :+1: The main change is not accepting `connect_kwargs` for the filename but as I commented already I think we need to add that.